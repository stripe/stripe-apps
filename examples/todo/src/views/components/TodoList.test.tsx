import { render } from '@stripe/ui-extension-sdk/testing';

import { List, ListItem } from '@stripe/ui-extension-sdk/ui';

import TodoList, { TodoListProps } from './TodoList';
import { Mode } from '../TodoApp';

describe('TodoList', () => {
  const mocks: TodoListProps = {
    todoList: [{
      text: 'foo',
      created: Date.now(),
      completed: false,
      notes: '',
    }],
    mode: Mode.Uncompleted,
    onDelete: jest.fn(),
    onComplete: jest.fn(),
    setOpenNotes: jest.fn(),
    setNotesTextFieldValue: jest.fn(),
  };

  it('renders the component correctly with todos', () => {
    const { wrapper } = render(<TodoList {...mocks} />);

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find(List)).toContainComponentTimes(ListItem, 1);
  });

  it('renders the component with no todos', () => {
    mocks.todoList = [];

    const { wrapper } = render(<TodoList {...mocks} />);

      expect(wrapper.find(List)).toContainComponentTimes(ListItem, 0);
  });
});