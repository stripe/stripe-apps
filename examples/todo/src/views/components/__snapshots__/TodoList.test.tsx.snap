// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders the component correctly with todos 1`] = `
Object {
  "act": [Function],
  "unmount": [Function],
}
`;
